### PR TITLE
Add Codecov Report Generation

### PR DESCRIPTION
**Pull request by gayaldassanayake**
$subject

### CODE REVIEW

File: transaction-ballerina/build.gradle [hunk 1]
```diff
@@ -119,8 +119,69 @@ task revertTomlFile {
...
+    }
+}
+
+test {
+    dependsOn(ballerinaTest)
+}
+
 task ballerinaBuild {
     dependsOn(":transaction-native:build")
+    dependsOn(initializeVariables)
```
Comment (manuranga) [inline transaction-ballerina/build.gradle:184]: In gradle it's a best practice to be declared. Therefore please don't put things into tasks unless each task is responsible for cohesive action or artifact. If you need a variable, in gradle it's better to initialize it within the task, and without having global variables.

**Review by manuranga — CHANGES_REQUESTED** [review]

File: transaction-ballerina/build.gradle [hunk 2]
```diff
@@ -135,9 +196,9 @@ task ballerinaBuild {
             workingDir project.projectDir
             environment "JAVA_OPTS", "-DBALLERINA_DEV_COMPILE_BALLERINA_ORG=true"
             if (Os.isFamily(Os.FAMILY_WINDOWS)) {
-                commandLine 'cmd', '/c', "$distributionBinPath/bal.bat build --skip-tests -c && exit %%ERRORLEVEL%%"
```
Comment (azinneera) [inline transaction-ballerina/build.gradle:138]: We should keep the `-c` flag

File: transaction-ballerina/build.gradle (hunk 1 above)
Comment (gayaldassanayake) [reply transaction-ballerina/build.gradle:184]: In that case `initializeVariables` should be removed right? So do we have to add that logic redundantly to both `ballerinaBuild` and `ballerinaTest`?

**Review by manuranga — APPROVED** [review]

File: transaction-ballerina/build.gradle [hunk 3]
```diff
@@ -152,15 +152,16 @@ task ballerinaTest {
     }
 
     def distributionBinPath =  project.projectDir.absolutePath + "/build/target/extracted-distributions/jballerina-tools-zip/jballerina-tools-${ballerinaLangVersion}/bin"
-
+    def testParams = "--code-coverage --includes=ballerinai.*:org.ballerinalang.stdlib.transaction.*"
```
Comment (Dilhasha) [inline transaction-ballerina/build.gradle:155]: We can move the testParams out of the task and re-use it in other tasks as well. WDYT?

File: codecov.yml [hunk 4]
```diff
@@ -3,5 +3,6 @@ fixes:
 
 ignore:
   - "**/tests"
```
Comment (Dilhasha) [inline codecov.yml:5]: We handle "**/tests" ignore internally now. You won't need this.

**Pull request merged**